test/spmd/test_tracing.py

```diff
@@ -380,6 +385,165 @@ def test_hybrid(self):
                 or p1.grad.allclose(p2.grad)
             )
 
+    @with_comms
```

Comment:
Agree - I've moved them out to test_graph_optimizations.py so that we have a central file for graph_optimizations test.  Can break it down to more specific ones (ala fusion, comm_overlapping, etc) in the future if this one gets too large in the future.